scipy/integrate/tests/test_integrate.py

```diff
@@ -716,6 +736,15 @@ def bjac_cols(y, t, c):
     assert_array_equal(info1['nje'], info2['nje'])
     assert_array_equal(info3['nje'], info4['nje'])
 
+    # Test the use of tfirst
+    sol1ty, info1ty = odeint(lambda t, y, c: func(y, t, c), y0, t, args=(c,),
+                             full_output=True, atol=1e-13, rtol=1e-11,
+                             mxstep=10000,
+                             Dfun=lambda t, y, c: jac(y, t, c), tfirst=True)
+    # The code should execute the exact same sequence of floating point
+    # calculations, so these should be exactly equal, and not just close.
```

Comment:
I'm not sure this statement is true: `func` calls `.dot` which is a BLAS routine, and those may not be floating-point exact reproducible because of e.g. internal threading, memory alignment, etc. issues.

Reply:
I guess if it is possible to call `odeint` twice with the exact same arguments consecutively in the same process and get different results, then it is certainly possible that this test could result in different answers.  (I'm curious to see if it ever happens, but that would mean that someday someone might report a test failure, and we would have to say "Yup, we knew that might happen in theory...")  I'll update the test.

Reply:
With LAPACK at least it's possible to hit this: https://github.com/scipy/scipy/issues/7878 
However, I'm not sure if small matrices + gemv can.